geode-core/src/integrationTest/java/org/apache/geode/cache/query/internal/QueryTraceJUnitTest.java

```diff
@@ -366,7 +396,11 @@ public void testTraceOnLocalRegionWithSmallTracePrefixNoComments() throws Except
     assertTrue(((DefaultQuery) query).isTraced());
 
     SelectResults results = (SelectResults) query.execute();
-    assertTrue(QueryObserverHolder.getInstance() instanceof IndexTrackingQueryObserver);
+
```

Comment:
These tests seem awfully repetitive. Is there a reason to check the test hook in every test?  From my perspective it seems like a waste of CPU. Just my opinion though.

Reply:
It is the way to check that the `trace` directive in the query has taken effect when set or that trace is not being done when it is not specified.